src/hardware/input/intel8042.cpp

```diff
@@ -892,6 +913,17 @@ static uint8_t read_data_port(io_port_t, io_width_t) // port 0x60
 		return data_byte;
 	}
 
+	if (is_diagnostic_dump && !buffer_num_used) {
+		// Diagnostic dump finished
+		is_diagnostic_dump = false;
```

Comment:
If the program doesn't read or drain the buffer, is there a chance we can get trapped in `is_diagnostic_dump`-mode?

Reply:
OK, should be correct now.